### PR TITLE
validate multiset dict counts

### PR DESCRIPTION
**Pull request by smichr**
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/utilities/iterables.py [hunk 1]
```diff
@@ -1365,11 +1365,15 @@ def multiset_combinations(m, n, g=None):
     """
     if g is None:
         if type(m) is dict:
-            if n > sum(m.values()):
+            if any(v < 0 for v in m.values()):
```
Comment (smichr) [inline sympy/utilities/iterables.py:1368]:
```suggestion
            if any(as_int(v) < 0 for v in m.values()):
```

File: sympy/utilities/iterables.py [hunk 2]
```diff
@@ -1411,6 +1418,8 @@ def multiset_permutations(m, size=None, g=None):
     """
     if g is None:
         if type(m) is dict:
+            if any(v < 0 for v in m.values()):
```
Comment (smichr) [inline sympy/utilities/iterables.py:1421]:
```suggestion
            if any(as_int(v) < 0 for v in m.values()):
```

File: sympy/utilities/iterables.py [hunk 3]
```diff
@@ -1392,7 +1399,7 @@ def multiset_combinations(m, n, g=None):
                     if len(rv) == n:
                         yield rv
 
-
+
```
Comment (smichr) [inline sympy/utilities/iterables.py:1402]:
```suggestion

```

File: sympy/utilities/iterables.py [hunk 4]
```diff
@@ -1365,11 +1365,15 @@ def multiset_combinations(m, n, g=None):
...
+            if any(as_int(v) < 0 for v in m.values()):
+                raise ValueError('counts cannot be negative')
+            N = sum(m.values())
+            if n > N:
                 return
             g = [[k, m[k]] for k in ordered(m)]
         else:
             m = list(m)
+            N = len(m)
             if n > len(m):
```
Comment (oscargus) [inline sympy/utilities/iterables.py:1377]: `if n > N:`?

**Pull request merged**